src/libopensc/pkcs15-pteid.c

```diff
@@ -106,22 +107,24 @@ static int sc_pkcs15emu_pteid_init(sc_pkcs15_card_t * p15card)
 	/* TODO: Use Portuguese descriptions? */
 	
 	/* Add X.509 Certificates */
-	for (i = 0; i < 4; i++) {
-		static const char *pteid_cert_names[4] = {
-				"AUTHENTICATION CERTIFICATE",
-				"SIGNATURE CERTIFICATE",
-				"SIGNATURE SUB CA",
-				"AUTHENTICATION SUB CA"
+	for (i = 0; i < 5; i++) {
+		static const char *pteid_cert_names[5] = {
+			"AUTHENTICATION CERTIFICATE",
+			"SIGNATURE CERTIFICATE",
+			"SIGNATURE SUB CA",
+			"AUTHENTICATION SUB CA",
+			"ROOT CA"
 		};
 		/* X.509 Certificate Paths */
-		static const char *pteid_cert_paths[4] = {
+		static const char *pteid_cert_paths[5] = {
 			"3f005f00ef09", /* Authentication Certificate path */
-			"3f005f00ef08", /* Digital Signature Certificate path */
+			"3f005f00ef08", /* Digital Signature Certificate path */			
 			"3f005f00ef0f", /* Signature sub CA path */
-			"3f005f00ef10"	/* Authentication sub CA path */
+			"3f005f00ef10",	/* Authentication sub CA path */
+			"3f005f00ef11"  /* Root CA path */
```

Comment:
If you add a comma after the "3f005f00ef11" string literal, when someone needs to add another value, they don't have to touch that line. (As you had to with "3f005f00ef10".) Sorry for not making this clearer.